### ISSUE TITLE
Snackbar not dismissed immediately after Dialog is Closed

### ISSUE THREAD

**Issue by teterovic** [angular/material2#6763]
#### Bug, feature request, or proposal:
When opening a snackbar after the close event of Dialog the snackbar is never dismissed

#### What is the expected behavior?
The snackbar should be dismissed/closed after the duration timout

#### What is the current behavior?
The snackbar never dismisses. When moved out of the subscribe block of the afterClosed function it works but thats not the intended behaviour 

#### What are the steps to reproduce?
```
        let dialogRef = this.dialog.open(DeleteChapterDialogComponent);
        dialogRef.afterClosed().subscribe((result: any) => {
            this.snackBar.open('LOL','OK', { duration: 100 });
        });
```
#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
    "@angular": "4.1.3",
    "@angular/material": "2.0.0-beta.8",

**Comment by josephperrott** [comment]
@teterovic can you provide a plunker with a reproduction?  

I was unable to reproduce this error myself on [plunker](http://plnkr.co/edit/GtRofySNILNX70b0qgfk?p=preview).

**Comment by teterovic** [comment]
@josephperrott I will try to update my project to Beta.10, i cant find a nice plunker with beta.8.
Either way it seems to be fixed in beta.10, but i will test it in my project too after the update maybe it's something else.

**Comment by teterovic** [comment]
Ok found the bug. Please replace
BrowserAnimationsModule with NoopAnimationsModule

Haven't tested it with beta.10, just beta 8

plunkr
https://plnkr.co/edit/x80EeFrLwq7naP1aVW5S?p=preview

**Comment by teterovic** [comment]
sorry this plunkr:
https://plnkr.co/edit/0UnXbq76xsaLBEyrEn8S?p=preview

**Comment by willshowell** [comment]
Maybe related to https://github.com/angular/material2/issues/6719?

**Comment by josephperrott** [comment]
@teterovic I was able to confirm that this is resolved with #6736.
